src/utils.ts

```diff
@@ -0,0 +1,72 @@
+import {closedShadowRootRegistry} from './shared';
+
+export var defineProperty = Object.defineProperty;
+export var forEach = Array.prototype.forEach;
+
+var importPattern = /@import.+?;?$/gm;
+
+export function rejectImports(contents: string): string {
+  var _contents = contents.replace(importPattern, '');
+
+  if (_contents !== contents) {
+    console.warn(
+      '@import rules are not allowed here. See https://github.com/WICG/construct-stylesheets/issues/119#issuecomment-588352418',
+    );
+  }
+
+  return _contents.trim();
+}
+
+/*#__PURE__*/
+export function clearRules(sheet: CSSStyleSheet): void {
+  while (sheet.cssRules.length > 0) {
+    sheet.deleteRule(0);
+  }
+}
+
+/*#__PURE__*/
+export function insertAllRules(from: CSSStyleSheet, to: CSSStyleSheet): void {
```

Comment:
Probably want to use this above instead of `this.insertRule`. I still wonder about the performance about these two and if it might cause unwanted flash of styles. Should this be in a `requestAnimationFrame`? Again, I'm mostly just thinking out loud. I will test these either this weekend or early next week.

Reply:
I tried both current and `requestAnimationFrame` approaches and don't see any specific difference in visualization and the flame chart. Could you test it as well? I think I would add `requestAnimationFrame` call anyway just to be sure it works at correct time.